lib/pushmi_pullyu/aip_downloader.rb

```diff
@@ -0,0 +1,147 @@
+require 'pushmi_pullyu/aip_downloader_files_directories'
+require 'pushmi_pullyu/aip_logger'
+require 'pushmi_pullyu/fedora_object_fetcher'
+require 'pushmi_pullyu/solr_fetcher'
+
+# Download all of the metadata/datastreams and associated data
+# related to an object
+
+class PushmiPullyu::AipDownloader
+
+  include PushmiPullyu::AipDownloaderFilesDirectories
+
+  attr_reader :noid, :config, :logger, :aip_logger, :fetcher,
+              :aipcreation_log
+
+  attr_accessor :got_characterization, :got_thumbnail,
+                :got_fedora3foxml, :got_fedora3foxml_metadata
+
+  def initialize(noid, config = nil, application_logger = nil)
+    @noid = noid
+
+    @config = config || PushmiPullyu.options
+    @logger = application_logger || PushmiPullyu.logger
+    @fetcher = PushmiPullyu::FedoraObjectFetcher.new(self.noid, self.config)
+
+    initialize_directory_names
+    initialize_filenames
+
+    # We need to log to application log and to the AIP log
+    @aip_logger = PushmiPullyu::AipLogger.new(@noid, aipcreation_log, @logger)
```

Comment:
It's logging both to the application log and to the AIP specific log. Maybe the hangup is that I spun the thing off to a class with "Logger" in it's name. All it's doing is writing log statements to both the application log and to the AIP creation log. I'm not sure about the multithread comment. While it's true that multiple threads could write to the same application logger, I don't think two threads would write to the AIP creation log.